crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2860,6 +2888,18 @@ impl KnownFunction {
     }
 }
 
+#[salsa::interned]
+pub struct ModuleLiteralType<'db> {
+    /// The file in which this module was imported.
+    ///
+    /// We need this in order to know which submodules should be attached to it as attributes
+    /// (because the submodules were also imported in this file).
+    pub importing_file: File,
```

Comment:
Does that mean we have multiple `ModuleLiteralType`s for the same module? One for each place from which it's imported?

Reply:
Yes that's right